internal/processors/cloudeventconvert/cloudeventconvert.go

```diff
@@ -165,37 +169,56 @@ func (c *cloudeventProcessor) processMsg(ctx context.Context, msg *service.Messa
 
 	retBatch, err := c.createEventMsgs(msg, source, hdrs, eventData)
 	if err != nil {
-		c.logger.Warn(fmt.Sprintf("failed to create event message: %s", contentType))
 		processors.SetError(msg, processorName, "failed to create event messages", err)
 		return service.MessageBatch{msg}
 	}
 	return retBatch
 }
 
-func (c *cloudeventProcessor) validateSignature(event *cloudevent.CloudEvent[json.RawMessage], attestor string) (bool, error) {
-	if !common.IsHexAddress(common.HexToAddress(attestor).Hex()) {
-		return false, errors.New("invalid attestor address")
-	}
-
+// verifySignature attempts to verify the signed data
+// first check if the source is the signer
+// if the source is not the signer, check whether the signature is from a dev license where the source is the contract addr
+func (c *cloudeventProcessor) verifySignature(event *cloudevent.CloudEvent[json.RawMessage], source common.Address) (bool, error) {
 	sig, ok := event.Extras["signature"].(string)
 	if !ok {
-		c.logger.Warn("failed to get signature from payload")
 		return false, errors.New("failed to get signed payload")
 	}
 
 	signature := common.FromHex(sig)
-	msgHash := crypto.Keccak256(event.Data)
-	pk, err := crypto.Ecrecover(msgHash, signature)
-	if err != nil {
-		return false, fmt.Errorf("failed to recover an recoveredPubKey: %w", err)
+	msgHash := crypto.Keccak256Hash(event.Data)
+	if len(signature) != 65 {
```

Comment:
Thing we noticed after merging: it's actually not okay to bail if this check fails, and we don't want to modify the `signature` we pass to the contract. Will fix ASAP.